zeebe/atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -104,10 +108,21 @@ public CompletableFuture<Void> stop() {
   }
 
   /** Truncates uncommitted entries from the log. */
-  private void truncateUncommittedEntries() {
-    if (role() == RaftServer.Role.PASSIVE && raft.getLog().getLastIndex() > raft.getCommitIndex()) {
-      raft.getLog().deleteAfter(raft.getCommitIndex());
-    }
+  private CompletableFuture<Void> truncateUncommittedEntries() {
+    final Runnable runnable =
+        () -> {
+          if (role() == RaftServer.Role.PASSIVE
+              && raft.getLog().getLastIndex() > raft.getCommitIndex()) {
+            try {
+              raft.getLog().deleteAfter(raft.getCommitIndex());
+            } catch (final FlushException e) {
+              LangUtil.rethrowUnchecked(e);
+            }
+          }
+        };
+    final var retryStrategy =
+        new ExponentialBackoffRetryDelay(Duration.ofSeconds(1), Duration.ofMillis(50));
+    return raft.getThreadContext().retryUntilSuccessful(runnable, retryStrategy);
```

Comment:
:thought_balloon: This sounds good in general - I'm wondering if I'm not forgetting a case where this could fail in an unrecoverable way, in which case we would want to go inactive/dead. :thinking: 

Maybe we can double check with Lena in case I'm forgetting anything.

Reply:
Maybe we want to really _only_ retry IOExceptions forever, but not retry anything else? :thinking:

Reply:
Yes I would say so as well

Reply:
So should we mark https://github.com/camunda/camunda/issues/24634 done as well?